python/cudf/cudf/core/column/decimal.py

```diff
@@ -224,6 +224,20 @@ def fillna(
 
         Returns a copy with null filled.
         """
+        if isinstance(value, (int, Decimal)):
+            value = pa.scalar(value, type=self.dtype.to_arrow()).as_py()
```

Comment:
I'm a little preferential to converting directly to `cudf.Scalar` and subsequently passing `device_value` at the cython interface. Right now in a lot of places in cython, there's a funnel (`as_device_scalar`) that will ultimately end up sending the resulting decimal here through a `cudf.Scalar` to obtain the final `DeviceScalar` anyways. But it would be nice to not have that, and just deal exclusively with `DeviceScalar`s in cython and `cudf.Scalar`s in python. 

It also makes it a little easier to spot when the host->device transfer happens (this happens when `device_value` is called) and ideally this would hit when we pass into the cython layer everywhere instead of possibly inside the cython somewhere.